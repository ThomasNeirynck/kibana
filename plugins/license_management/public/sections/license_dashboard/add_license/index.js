export { AddLicense } from './add_license';
