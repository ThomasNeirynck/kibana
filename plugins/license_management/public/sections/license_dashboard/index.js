export { LicenseDashboard } from './license_dashboard';
