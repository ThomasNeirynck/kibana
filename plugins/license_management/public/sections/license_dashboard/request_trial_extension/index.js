export { RequestTrialExtension } from './request_trial_extension.container';
