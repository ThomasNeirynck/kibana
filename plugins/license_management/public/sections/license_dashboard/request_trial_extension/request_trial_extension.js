import React from 'react';

import { EuiFlexItem, EuiCard, EuiLink, EuiButton } from '@elastic/eui';

export const RequestTrialExtension = ({ shouldShowRequestTrialExtension }) => {
  if (!shouldShowRequestTrialExtension) {
    return null;
  }
  const description = (
    <span>
      If youâ€™d like to continuing using Security, Machine Learning, and our
      other awesome{' '}
      <EuiLink
        href="https://www.elastic.co/subscriptions/xpack"
        target="_blank"
      >
        platinum features
      </EuiLink>, request an extension now.
    </span>
  );
  return (
    <EuiFlexItem>
      <EuiCard
        title="Extend your trial"
        description={description}
        footer={
          <EuiButton
            style={{ marginTop: 'auto' }}
            target="_blank"
            href="https://www.elastic.co/trialextension"
          >
            Extend trial
          </EuiButton>
        }
      />
    </EuiFlexItem>
  );
};
