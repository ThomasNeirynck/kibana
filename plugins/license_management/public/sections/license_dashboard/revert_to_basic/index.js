export { RevertToBasic } from './revert_to_basic.container';
