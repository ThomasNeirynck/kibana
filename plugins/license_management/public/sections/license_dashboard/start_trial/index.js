export { StartTrial } from './start_trial.container';