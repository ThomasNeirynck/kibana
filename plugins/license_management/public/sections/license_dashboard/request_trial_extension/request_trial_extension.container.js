import { connect } from 'react-redux';

import { RequestTrialExtension as PresentationComponent } from './request_trial_extension';
import {
  shouldShowRequestTrialExtension
} from '../../../store/reducers/licenseManagement';


const mapStateToProps = state => {
  return {
    shouldShowRequestTrialExtension: shouldShowRequestTrialExtension(state),
  };
};

export const RequestTrialExtension = connect(mapStateToProps)(
  PresentationComponent
);
