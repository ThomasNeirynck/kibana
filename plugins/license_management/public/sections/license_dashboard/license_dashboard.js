import React from 'react';


import { LicenseStatus } from './license_status';
import { RevertToBasic } from './revert_to_basic';
import { StartTrial } from './start_trial';
import { AddLicense } from './add_license';
import { RequestTrialExtension } from './request_trial_extension';
import {
  EuiFlexGroup,
  EuiFlexItem,
  EuiSpacer
} from '@elastic/eui';

export const LicenseDashboard = () => {
  return (
    <div className="licenseManagement__contain">
      <EuiFlexGroup justifyContent="spaceAround">
        <LicenseStatus />
      </EuiFlexGroup>
      <EuiSpacer size="l" />
      <EuiFlexGroup justifyContent="spaceAround">
        <EuiFlexItem>
          <AddLicense />
        </EuiFlexItem>
        <StartTrial />
        <RequestTrialExtension/>
        <RevertToBasic/>
      </EuiFlexGroup>
    </div>
  );
};
