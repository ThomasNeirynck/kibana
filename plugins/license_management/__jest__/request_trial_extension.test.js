import { RequestTrialExtension } from '../public/sections/license_dashboard/request_trial_extension';
import { createMockLicense, getComponent } from './util';

describe('RequestTrialExtension component', () => {
  test('should not display when trial expires in > 24 days', () => {
    const nonImminentExpirationTime = new Date().getTime() + (30 * 24 * 60 * 60 * 1000);
    const rendered = getComponent(
      {
        license: createMockLicense('trial', nonImminentExpirationTime)
      },
      RequestTrialExtension
    );
    expect(rendered.html()).toMatchSnapshot();
  });
  test('should display when trial license is expired', () => {
    const rendered = getComponent(
      {
        license: createMockLicense('trial', 0)
      },
      RequestTrialExtension
    );
    expect(rendered.html()).toMatchSnapshot();
  });
  test('should display when trial license is about to expire', () => {
    // ten days from now
    const imminentExpirationTime = new Date().getTime() + (10 * 24 * 60 * 60 * 1000);
    const rendered = getComponent(
      {
        license: createMockLicense('trial', imminentExpirationTime)
      },
      RequestTrialExtension
    );
    expect(rendered.html()).toMatchSnapshot();
  });
  test('should not display for active basic license', () => {
    const rendered = getComponent(
      {
        license: createMockLicense('basic')
      },
      RequestTrialExtension
    );
    expect(rendered.html()).toBeNull();
  });
  test('should not display for active gold license', () => {
    const rendered = getComponent(
      {
        license: createMockLicense('gold')
      },
      RequestTrialExtension
    );
    expect(rendered.html()).toBeNull();
  });
  test('should not display for active platinum license', () => {
    const rendered = getComponent(
      {
        license: createMockLicense('platinum')
      },
      RequestTrialExtension
    );
    expect(rendered.html()).toBeNull();
  });
});
