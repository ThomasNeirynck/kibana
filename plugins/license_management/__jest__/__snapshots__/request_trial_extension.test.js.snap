// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RequestTrialExtension component should display when trial license is about to expire 1`] = `null`;

exports[`RequestTrialExtension component should display when trial license is expired 1`] = `null`;

exports[`RequestTrialExtension component should not display when trial expires in > 24 days 1`] = `null`;
